osu.Game/Overlays/Comments/CommentsContainer.cs

```diff
@@ -0,0 +1,198 @@
+                                        },
+                                        new Container
+                                        {
+                                            AutoSizeAxes = Axes.Y,
+                                            RelativeSizeAxes = Axes.X,
+                                            Child = moreButton = new CommentsShowMoreButton
+                                            {
+                                                Anchor = Anchor.Centre,
+                                                Origin = Anchor.Centre,
+                                                Margin = new MarginPadding(5),
+                                                Action = () => getComments(false),
+                                            }
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            });
+        }
+
+        protected override void LoadComplete()
+        {
+            Sort.BindValueChanged(onSortChanged, true);
+            base.LoadComplete();
+        }
+
+        private void onSortChanged(ValueChangedEvent<CommentsSortCriteria> sort) => getComments();
+
+        private void getComments(bool initial = true)
+        {
+            if (initial)
+            {
+                currentPage = 1;
+                loadedTopLevelComments = 0;
+                deletedChildrenPlaceholder.DeletedCount.Value = 0;
+                moreButton.IsLoading = true;
+                content.Clear();
+            }
+
+            request?.Cancel();
+            loadCancellation?.Cancel();
+            request = new GetCommentsRequest(type, id, Sort.Value, currentPage++);
+            request.Success += response => onSuccess(response, initial);
+            api.Queue(request);
+        }
+
+        private void onSuccess(APICommentsController response, bool initial)
+        {
+            loadCancellation = new CancellationTokenSource();
+
+            FillFlowContainer page = new FillFlowContainer
+            {
+                RelativeSizeAxes = Axes.X,
+                AutoSizeAxes = Axes.Y,
+                Direction = FillDirection.Vertical,
+            };
+
+            foreach (var c in response.Comments)
+            {
+                if (c.IsTopLevel)
+                    page.Add(new DrawableComment(c)
+                    {
+                        ShowDeleted = { BindTarget = ShowDeleted }
+                    });
+            }
+
+            LoadComponentAsync(page, loaded =>
+            {
+                content.Add(loaded);
+
+                int deletedComments = response.Comments.Select(c => c.IsDeleted && c.IsTopLevel).Count(c => c);
+
+                deletedChildrenPlaceholder.DeletedCount.Value = initial ? deletedComments : deletedChildrenPlaceholder.DeletedCount.Value + deletedComments;
```

Comment:
```suggestion
                deletedChildrenPlaceholder.DeletedCount.Value += deletedComments;
```

This `initial` bool can be removed